src/bucketblocker/main.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/sts"
+)
+
+func blockPublicAccess(name string, s3Client s3.S3) (*s3.PutPublicAccessBlockOutput, error) {
+	resp, err := s3Client.PutPublicAccessBlock(&s3.PutPublicAccessBlockInput{
+		Bucket: aws.String(name),
+		PublicAccessBlockConfiguration: &s3.PublicAccessBlockConfiguration{
+			BlockPublicAcls:       aws.Bool(true),
+			IgnorePublicAcls:      aws.Bool(true),
+			BlockPublicPolicy:     aws.Bool(true),
+			RestrictPublicBuckets: aws.Bool(true),
+		},
+	})
+	if err != nil {
+		return resp, err
+	}
+	fmt.Println("Public access blocked for bucket: " + name)
+	return resp, nil
+}
+
+func validateCredentials(stsClient sts.STS, profile string) (*sts.GetCallerIdentityOutput, error) {
+	resp, err := stsClient.GetCallerIdentity(&sts.GetCallerIdentityInput{})
+	if err != nil {
+		return resp, errors.New("Could not find valid credentials for profile: " + profile)
+	}
+	return resp, nil
+}
+
+func main() {
+	name := flag.String("bucket", "", "The name of the bucket to block")
+	profile := flag.String("profile", "default", "The name of the profile to use")
+	region := flag.String("region", "eu-west-1", "The region of the bucket")
+	flag.Parse()
+
```

Comment:
Could probably fail early here if the bucket name is empty? Otherwise we're making a bunch of calls to AWS for no reason.